torchviz/dot.py

```diff
@@ -3,71 +3,164 @@
 from graphviz import Digraph
 import torch
 from torch.autograd import Variable
+import warnings
 
 Node = namedtuple('Node', ('name', 'inputs', 'attr', 'op'))
 
-
-def make_dot(var, params=None):
+# Saved attrs for grad_fn (incl. saved variables) begin with `._saved_*`
+SAVED_PREFIX = "_saved_"
+
+def get_fn_name(fn, show_attrs, max_attr_chars):
+    name = str(type(fn).__name__)
+    if not show_attrs:
+        return name
+    attrs = dict()
+    for attr in dir(fn):
+        if not attr.startswith(SAVED_PREFIX):
+            continue
+        val = getattr(fn, attr)
+        attr = attr[len(SAVED_PREFIX):]
+        if torch.is_tensor(val):
+            attrs[attr] = "[saved tensor]"
+        elif isinstance(val, tuple) and any(torch.is_tensor(t) for t in val):
+            attrs[attr] = "[saved tensors]"
+        else:
+            attrs[attr] = str(val)
+    if not attrs:
+        return name
+    max_attr_chars = max(max_attr_chars, 3)
```

Comment:
Why do you put this minimum of 3 for the width here?

Reply:
I want to add '...' to indicate that an entry has been truncated, but '...' itself has 3 characters. So this avoids some issues with people setting values of max_attr_chars less than 3.

Reply:
Sounds good!